Adding an 'tested-by' flag to the test report. (#317)
w3c/epub-tests: CONTRIBUTING.md

* `ref` (optional): A URL that creates a link on the name of the reading system in the implementation report.

> * `official` (optional): If set to `true`, the tests have been performed by a persons that officially represent the implementers of the reading system for the purpose of testing.

Reviewer: ```suggestion
* `tested-by` (optional): If set to `implementer` the tests have been performed by the implementers of the reading system, otherwise it should be set to `third-party`.
```

Leaving this vague, since there can be a lot of people we might consider implementers (e.g. significant contributor to an open source project).

* `tests`: An object with the list of the implementation results. Each key is a test's unique identifier (its `dc:identifier`) with a
  value of `true`, `false`, `"n/a"`, or `null` for a test that passes, fails, is not applicable (i.e., is not implemented for some reasons), or is not yet tested, respectively. If a test is not listed, its value is considered to be `null`. The implementation report will show "?" for `null`, indicating that the implementation has not run the test.

Here is an example of a small test report:

```json
{
    "name"     : "ACME Books",
    "ref"      : "https://www.example.org/acme",
    "variant"  : "iOS, v1.0",

>     "official" : true,

Reviewer: ```suggestion
    "tested-by" : "implementer",
```

Reviewer 2: Would this be `"tested_by" : "implementer",` if your other changes are accepted?

Reviewer: It would. My suggestions (here and elsewhere) all use the current code in case there is a reason not to take take the other suggestion (hyphen to underscore).

Author: But I may prefer keeping this with '-'. See https://github.com/w3c/epub-tests/pull/317#discussion_r3676037820.

I will accept this change because it is a bug. We can update later.
